static/js/stream_popover.js

```diff
@@ -293,8 +293,47 @@ function build_move_topic_to_stream_popover(e, current_stream_id, topic_name) {
         ui_util.decorate_stream_bar(stream_name, stream_header_colorblock, false);
     });
 
-    $("#move_topic_modal").modal("show");
-    e.stopPropagation();
+    // The API endpoint for editing messages to change their
+    // content, topic, or stream requires a message ID.
+    //
+    // Because we don't have full data in the browser client, it's
+    // possible that we might display a topic in the left sidebar
+    // (and thus expose the UI for moving its topic to another
+    // stream) without having a message ID that is definitely
+    // within the topic.  (The comments in stream_topic_history.js
+    // discuss the tricky issues around message deletion that are
+    // involved here).
+    //
+    // To ensure this option works reliably at a small latency
+    // cost for a rare operation, we just ask the server for the
+    // latest message ID in the topic.
+    const data = {
+        anchor: "newest",
+        num_before: 1,
+        num_after: 0,
+        narrow: JSON.stringify([
+            {operator: "stream", operand: current_stream_id},
+            {operator: "topic", operand: topic_name},
+        ]),
+    };
+
+    channel.get({
+        url: "/json/messages",
+        data,
+        idempotent: true,
+        success(data) {
+            $("#move_topic_modal").data("message-id", data.messages[0].id);
```

Comment:
We generally avoid using jQuery's `.data()` as a setter because if you try to read it with `.attr('data-message-id)`, it doesn't work.